sunpy/net/dataretriever/client.py

```diff
@@ -225,7 +176,63 @@ def _can_handle_query(cls, *query):
         `sunpy.net.fido_factory.UnifiedDownloaderFactory`
         class uses to dispatch queries to this Client.
         """
-        raise NotImplementedError
+        from sunpy.net import attrs as a
+
+        required = {a.Time, a.Instrument}
+        adict = cls.register_values()
+        optional = {i for i in adict.keys()} - required
+        if hasattr(cls, 'required'):
+            required = cls.required
+        if hasattr(cls, 'optional'):
+            optional = cls.optional
+        if not cls.check_attr_types_in_query(query, required, optional):
```

Comment:
As per the thread above:
```suggestion
        if not cls.check_attr_types_in_query(query, cls.required, cls.optional):
```

Reply:
Done